packages/api/src/external/commonwell/document/shared.ts

```diff
@@ -39,3 +48,46 @@ export function getFileExtension(value: string | undefined): string {
   const fileExtension = extension(value);
   return fileExtension ? `.${fileExtension}` : "";
 }
+
+export function getExtraDocRefInfo(content: DocumentContent | undefined): DocumentReferenceExtras {
+  const extraInfo: DocumentReferenceExtras = {};
+  const contained = content?.contained;
+  const context = content?.context;
+
+  if (contained) {
+    contained.forEach(item => {
+      const containedName = item.name;
+      if (item.resourceType === "Organization") {
+        if (typeof containedName === "string") {
+          extraInfo.organization = {
+            name: containedName,
+          };
+        }
+      } else if (item.resourceType === "Practitioner" && containedName) {
+        if (Array.isArray(containedName)) {
+          const name = containedName[0];
+          if (name) addPractitionerName(extraInfo, name);
+        } else if (typeof containedName === "object") {
+          addPractitionerName(extraInfo, containedName);
+        }
+      }
+    });
+  }
+
+  if (context?.facilityType?.text) {
+    extraInfo.facilityType = context.facilityType.text;
+  }
+  return extraInfo;
+}
+
+export function addPractitionerName(extraInfo: DocumentReferenceExtras, name: HumanName): void {
+  if (name && name.given?.length && name.given[0] != "") {
+    extraInfo.practitioner = {
+      name: {
+        given: name.given,
+        family: name.family,
+        prefix: name.prefix?.length && name.prefix[0] !== "" ? name.prefix : undefined,
```

Comment:
> `prefix[0]`

what if there's more than one element in the array?

Reply:
if there's more than one, we'll get all of them. I'm just filtering out this case: `prefix: [""]`

Reply:
Sorry, I meant the fact we're referring to the first item of the array here but also on the `if` on line 84; I chose this line to comment thinking they were referring to the same thing when I quickly scanned the code 😄 .